user/models.go

```diff
@@ -69,6 +69,7 @@ type User struct {
 	Deactivated       bool                   `default:"false" bson:"deactivated" json:"deactivated"`
 	DeactivatedAt     time.Time              `bson:"deactivated_at" json:"deactivated_at"`
 	IsVerified        bool          		 `bson:"isverified" json:"isverified"`
+	MailingList       bool          		 `bson:"mailing_list" json:"mailing_list"`
```

Comment:
Mailing list shouldn't be a field under user. We would have separate collection to handle users that what to receive zuri mails. It's not a field under user. 

And why is it a boolean field and yet a "MailingList"?

Reply:
My thought process is that....since we alteady have signed up users with verified mails

We just go on further to have a yes or no field if they'll like to get mails

If yes, we write a function that searches through the user database and returns the email of users who opted for zuri mails

Reply:
I'll extend it to create a collection for the subscribed mails...but I'll still need to be able to verify mails and those who've subscribed before. Hence I'll still need a boolean field on the struct unless I find a way to work around codei8's codes plus mail credentials needed

Reply:
I think you are complicating this thing. Here's what I think. I believe there is a way users determine if they want to receive promo mails and all of that. Simply curate a list of those that opted to receive mail. You do not need to verify them. A total stranger may read the blog and be interested in receiving promo mails. He'll be requested to enter  just his email. It doesn't necessarily mean he MUST be an account holder in that platform. It's just mails he's interested in. You get my point?

So it's like we have collection MailingList. And store emails of the subscribers there. With few details, email, date the subscribed, subscribed (bool field we toggle incase they want to opt out in the future) When we want to send mails we fetch the emails in the collection and do a bulk emailing to all of them. 

This is easily and less expensive than running through the entire user collection to find who is interested in our mails or not. In fact to reduce run time, we can even discard the email from the collection when they unsubscribe so we do not need the "subscribed" bool field.

Reply:
Alright Boss

Once I get out of Interimo I'll push something to work this way